Bluetooth: BR: Format BR shell bredr.c

lylezhu2012 opened the pull request:
Change `int res` to `int err`.

Return error code if the command is failed instead of `0`.

Return `SHELL_CMD_HELP_PRINTED` if `shell_help` is called.

Remove duplicate `else` statement if the function is returned from corresponding `if` statement.

lylezhu2012 commented:
@jhedberg @carlescufi When I just resolved this PR conflict, I found that I didn't receive any notifications for some changes that might be related to BR. There have been many similar situations before. So I want to ask if I can automatically get notifications when there are changes in the common part or the br part.

jhedberg commented:
> Change `int res` to `int err`.
> 
> Return error code if the command is failed instead of `0`.
> 
> Return `SHELL_CMD_HELP_PRINTED` if `shell_help` is called.
> 
> Remove duplicate `else` statement if the function is returned from
corresponding `if` statement.

All of those could be in four separate commits. This helps review, easier mapping of changes to commit message (explanation of the changes), and it also helps bisectability later. Additionally, it boosts your contributor statistics :)

lylezhu2012 committed:
Bluetooth: BR: Return `SHELL_CMD_HELP_PRINTED` if calls shell_help

If the function `shell_help` is called, return SHELL_CMD_HELP_PRINTED
instead of `0`.

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>

lylezhu2012 committed:
Bluetooth: BR: shell: Rename `int res` to `int err`

Rename `int res` to `int err`. Make variable naming consistent.

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>

lylezhu2012 committed:
Bluetooth: BR: Shell: Return error if command is failed

Return error code if the command is failed instead of `0`.

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>

lylezhu2012 committed:
Bluetooth: BR: Shell: Remove duplicated `else` statement

Remove duplicate `else` statement if the function is returned from
corresponding `if` statement.

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>

jhedberg commented:
> When I just resolved this PR conflict, I found that I didn't receive any notifications for some changes that might be related to BR. There have been many similar situations before. So I want to ask if I can automatically get notifications when there are changes in the common part or the br part.

This happens completely based on `MAINTAINERS.yml`. I'd suggest to take a look there to see if we can improve how BR/EDR vs common vs LE parts are specified. The granularity is on a per-file basis however, so in some cases it may be worth considering if there's a need to do some refactoring, so that BR/EDR code gets moved out to a separate file.
